samples/news/News.UITests/Pages/SourcesPage.cs

```diff
@@ -9,8 +15,28 @@ public class SourcesPage : BasePage
             iOS = x => x.Marked(nameof(SourcesPage))
         };
 
+        protected readonly Query SourceLabel;
+
+        public List<string> SupportedSources => Constants.Sources.All;
+
         public SourcesPage()
         {
+            SourceLabel = x => x.Marked(nameof(SourceLabel));
+        }
+
+        public void ShowNextSource()
```

Comment:
why is the return type `void`? we aren't leaving the SourcePage (Tab) right? we stay in the same tab? this should return SourcePage. 

`void` is for navigation away from the current page, or in the case of this app, navigation to a different tab

Reply:
It was done by following the example in the `AddFavorite` test (which was already implemented earlier and served as an architectural decision for me):

```
   public void AddFavorite()
        {
            app.WaitForElement(FavoriteButton);
            app.Tap(FavoriteButton);
            app.Screenshot("Added favorite article");
        }
```

Here we being at the News page adding to favorites, no navigation occurs, the return type is this void.

Reply:
Let me update all the test to reflect the `navigation away` logic.